src/databricks/labs/dqx/col_functions.py

```diff
@@ -281,20 +281,50 @@ def not_greater_than(col_name: str, limit: int | datetime.date | datetime.dateti
     )
 
 
+def _get_min_max_column_expr(
+    min_limit: int | datetime.date | datetime.datetime | str | None = None,
+    max_limit: int | datetime.date | datetime.datetime | str | None = None,
+    min_limit_col_expr: str | Column | None = None,
+    max_limit_col_expr: str | Column | None = None,
+) -> tuple[Column, Column]:
+    """Helper function to create a condition for the is_(not)_in_range functions.
+
+    :param min_limit: min limit value
+    :param max_limit: max limit value
+    :param min_limit_col_expr: min limit column name or expr
+    :param max_limit_col_expr: max limit column name or expr
+    :return: tuple containing min_limit_expr and max_limit_expr
+    """
+    if min_limit_col_expr is None:
+        min_limit_expr = F.lit(min_limit)
+    else:
+        min_limit_expr = F.col(min_limit_col_expr) if isinstance(min_limit_col_expr, str) else min_limit_col_expr
+    if max_limit_col_expr is None:
+        max_limit_expr = F.lit(max_limit)
+    else:
+        max_limit_expr = F.col(max_limit_col_expr) if isinstance(max_limit_col_expr, str) else max_limit_col_expr
```

Comment:
we need to throw an error if both `min_limit` and `min_limit_col_expr` are `None` (same for `max_limit` and `max_limit_col_expr`).

Otherwise we'll get conditions like `(F.col(col_name) < F.lit(None)) | (F.col(col_name) > F.lit(None))`